lib/iris/tests/unit/common/metadata/test_CellMeasureMetadata.py

```diff
@@ -58,10 +62,570 @@ def test__fields(self):
+    def setUp(self):
+        self.values = dict(
+            standard_name=sentinel.standard_name,
+            long_name=sentinel.long_name,
+            var_name=sentinel.var_name,
+            units=sentinel.units,
+            attributes=sentinel.attributes,
+            measure=sentinel.measure,
+        )
+        self.dummy = sentinel.dummy
+        self.cls = CellMeasureMetadata
+        self.none = self.cls(*(None,) * len(self.cls._fields))
+
+    def test_wraps_docstring(self):
+        self.assertEqual(
+            BaseMetadata.combine.__doc__, self.cls.combine.__doc__,
+        )
+
+    def test_lenient_service(self):
+        qualname_combine = _qualname(self.cls.combine)
+        self.assertIn(qualname_combine, _LENIENT)
+        self.assertTrue(_LENIENT[qualname_combine])
+        self.assertTrue(_LENIENT[self.cls.combine])
+
+    def test_lenient_default(self):
+        other = sentinel.other
+        return_value = sentinel.return_value
+        with mock.patch.object(
+            BaseMetadata, "combine", return_value=return_value
+        ) as mocker:
+            result = self.none.combine(other)
+
+        self.assertEqual(return_value, result)
+        self.assertEqual(1, mocker.call_count)
+        (arg,), kwargs = mocker.call_args
+        self.assertEqual(other, arg)
+        self.assertEqual(dict(lenient=None), kwargs)
+
+    def test_lenient(self):
+        other = sentinel.other
+        lenient = sentinel.lenient
+        return_value = sentinel.return_value
+        with mock.patch.object(
+            BaseMetadata, "combine", return_value=return_value
+        ) as mocker:
+            result = self.none.combine(other, lenient=lenient)
+
+        self.assertEqual(return_value, result)
+        self.assertEqual(1, mocker.call_count)
+        (arg,), kwargs = mocker.call_args
+        self.assertEqual(other, arg)
+        self.assertEqual(dict(lenient=lenient), kwargs)
+
+    def test_op_lenient_same(self):
+        lmetadata = self.cls(**self.values)
+        rmetadata = self.cls(**self.values)
+        expected = self.values
+
+        with mock.patch("iris.common.metadata._LENIENT", return_value=True):
+            self.assertEqual(expected, lmetadata.combine(rmetadata)._asdict())
+            self.assertEqual(expected, rmetadata.combine(lmetadata)._asdict())
+
+    def test_op_lenient_same_none(self):
+        lmetadata = self.cls(**self.values)
+        right = self.values.copy()
+        right["var_name"] = None
+        rmetadata = self.cls(**right)
+        expected = self.values
+
+        with mock.patch("iris.common.metadata._LENIENT", return_value=True):
+            self.assertEqual(expected, lmetadata.combine(rmetadata)._asdict())
+            self.assertEqual(expected, rmetadata.combine(lmetadata)._asdict())
+
+    def test_op_lenient_same_measure_none(self):
+        lmetadata = self.cls(**self.values)
+        right = self.values.copy()
+        right["measure"] = None
```

Comment:
I don't think it's actually possible to have a CellMeasure with a measure of None. In the `CellMeasure.__init__` if no measure is given, it sets it to be `area` (although this is quite strange behaviour in itsefl!) and the `measure.setter` only allows `area` or `volume`
Although it seems possible to create a CellMeasureMetadata with measure = None, in which case it is probably safest to still test this

Reply:
I think this is still a valid test. Not that there is no sense of the logic applied by the `mixin` or the `CellMeasure` class itself here - we're disjointed from that, so we're actually free to set the member values to absolutely anything that we want. 

That's totally fine. It's the higher order logic that verifies the values; that contract doesn't belong within any of the namedtuple metadata classes.

So in reality this test is just ensuring that there is nothing exceptional happening with `None`, and it behaves as we expect in a lenient way.

I suspect that by enforcing this common treatment and approach across `iris` in a wholistic fashion (as we're doing in this series of PRs) will weed out some pretty interesting upper level oddities and questions ... such as why can't the `measure = None`?

Note that, another tangible example such an outcome from this is us now applying a little more rigour to the default `units` of our metadata. FTW